equinox/nn/_linear.py

```diff
@@ -23,6 +23,7 @@ def __init__(
         in_features: Union[int, Literal["scalar"]],
         out_features: Union[int, Literal["scalar"]],
         use_bias: bool = True,
+        dtype=jnp.float32,
```

Comment:
Shouldn't the default dtype be dependent on whatever this is set to:

`jax.config.x64_enabled`

?